src/dailyai/services/ai_services.py

```diff
@@ -109,12 +109,20 @@ def __init__(self, **kwargs):
 
     @abstractmethod
     async def run_vision(self, frame: VisionImageFrame) -> str:
+        print(f"### uh oh, abstract run vision")
         pass
 
     async def process_frame(self, frame: Frame) -> AsyncGenerator[Frame, None]:
+        print(f"### visionservice process frame, {frame}")
         if isinstance(frame, VisionImageFrame):
-            description = await self.run_vision(frame)
-            yield TextFrame(description)
+            print(f"### ### calling self.run_vision")
+            if getattr(self, "run_vision_async"):
+                async for frame in self.run_vision_async(frame):
+                    yield frame
+
+            else:
+                description = await self.run_vision(frame)
+                yield TextFrame(description)
```

Comment:
shouldn't `run_vision` be async already?

Reply:
It is, but naming things is hard 😂 the way you implemented the `run_vision` method for moondream returns a string that you wrap in a text frame, but the the way the OpenAI module returns completions is an AsyncGenerator, so I needed another `run_vision` method. It's clunky but I'm out of ideas

Reply:
I think we should make Moondream return AsyncGenerator :-). I'll fix that tomorrow.